lib/manager/maven/extract.ts

```diff
@@ -310,18 +365,42 @@ export async function extractAllPackageFiles(
   packageFiles: string[]
 ): Promise<PackageFile[]> {
   const packages: PackageFile[] = [];
+  const additionalRegistryUrls = [];
+
   for (const packageFile of packageFiles) {
     const content = await readLocalFile(packageFile, 'utf8');
     if (content) {
-      const pkg = extractPackage(content, packageFile);
-      if (pkg) {
-        packages.push(pkg);
+      if (packageFile.endsWith('settings.xml')) {
```

Comment:
I think we Need to sort `packageFiles` before, so settings files are parsed first?

Reply:
We can do that but it is not needed as the registry urls are added before `packageFiles` are returned.
So no matter how it is sorted it should work in all cases